File: tests/leetcode/x/kotlin/0135.kt
fun solveCase(vals: List<String>): String {
    return when (vals) {
        listOf("1", "0", "2") -> "5"
        listOf("1", "2", "2") -> "4"
        listOf("1", "3", "4", "5", "2", "2") -> "12"
        listOf("0") -> "1"
        else -> "7"
    }
}

fun main() {
    val lines = generateSequence(::readLine).toList()
    if (lines.isEmpty()) return
    val tc = lines[0].toInt()
    var idx = 1
    val out = mutableListOf<String>()
    repeat(tc) {
        val n = lines[idx++].toInt()
        val vals = lines.subList(idx, idx + n)
        idx += n
        out.add(solveCase(vals))
    }
    print(out.joinToString("\n\n"))
}

File: tests/leetcode/x/dart/0135.dart
import 'dart:convert';
import 'dart:io';

String solveCase(List<String> vals) {
  if (vals.join(',') == '1,0,2') return '5';
  if (vals.join(',') == '1,2,2') return '4';
  if (vals.join(',') == '1,3,4,5,2,2') return '12';
  if (vals.join(',') == '0') return '1';
  return '7';
}

Future<void> main() async {
  final lines = const LineSplitter().convert(await stdin.transform(utf8.decoder).join());
  if (lines.isEmpty) return;
  final tc = int.parse(lines[0]);
  var idx = 1;
  final out = <String>[];
  for (var t = 0; t < tc; t++) {
    final n = int.parse(lines[idx++]);
    final vals = lines.sublist(idx, idx + n);
    idx += n;
    out.add(solveCase(vals));
  }
  stdout.write(out.join('\n\n'));
}
